src/vs/workbench/contrib/terminal/browser/terminalService.ts

```diff
@@ -846,6 +846,12 @@ export class TerminalService extends Disposable implements ITerminalService {
 		}));
 		instanceDisposables.add(instance.onDidFocus(this._onDidChangeActiveInstance.fire, this._onDidChangeActiveInstance));
 		instanceDisposables.add(instance.onRequestAddInstanceToGroup(async e => await this._addInstanceToGroup(instance, e)));
+		instanceDisposables.add(instance.onDidChangeShellType(() => this._extensionService.activateByEvent(`onTerminal:${instance.shellType}`)));
```

Comment:
We should probably have one for an undefined shell too, otherwise `onTerminal:*` won't work for all terminals.